editor/state.js

```diff
@@ -91,13 +96,28 @@ export const blocks = combineUndoableReducers( {
 					...state.slice( index + 2 )
 				];
 
+			case 'REPLACE_BLOCKS':
+				if ( ! action.blocks ) {
+					return state;
+				}
+				index = state.indexOf( action.uids[ 0 ] );
+				return state.reduce( ( memo, uid ) => {
+					if ( uid === action.uids[ 0 ] ) {
```

Comment:
I see the need to support replacing multiple blocks, but it feels strange that we necessarily assume that if there are multiple blocks to be replaced, they're guaranteed to be in consecutive order. Do you think it would be any better in `REPLACE_BLOCKS` that we only specify a single `uid` to be replaced, and in the `block.js` we call to `REMOVE_BLOCK` the remainder of the blocks to be replaced?